### PR TITLE
Speed up testing and switch to stronger-typed asserts

### PR DESCRIPTION
**Pull request by janbridley**
## Description

Testing prior to this PR was slow, required reruns, and used base python asserts that make debugging output difficult. These issues have been addressed:

- Previously, tests that fail with the default config ran a random sample optimizer. This has been switched to the quaternion lattice generator, which samples SO(3) more efficiently.
- Free parameters in the testing step gradient descent optimizer have been tuned. While the new variant converges more slowly, it finds much better minima.
- Reruns have been removed, as the tuned optimizers now easily converge to the required tolerance.
- Base python asserts are now done with np.testing where possible, and the shape of outputs are verified as well as the value. This will catch errors where the incorrect number of points are returned from C.
- Global variables in test_pgop.py are now capitalized, making the inputs more distinct.
- The global RTOL for pgop tests (1e-4) is now a variable at the top of the file

## Motivation and Context

Resolves: #???

## Checklist:
- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/spatula/blob/main/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**spatula Contributor Agreement**](https://github.com/glotzerlab/spatula/blob/main/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/spatula/blob/main/AUTHORS.rst) in the pull request source branch.
- [ ] I have updated the [Change log](https://github.com/glotzerlab/spatula/blob/main/changelog.rst).

### CODE REVIEW

File: tests/test_pgop.py [hunk 1]
```diff
@@ -1510,17 +1510,25 @@ def compute_pgop_polyhedron(
...
+    )
+    op_pg = compute_op_result(symmetry, OPTIMIZER, mode, system, nlist, sigma, qp)
+    assert op_pg.order.shape == expected_shape
     if not np.allclose(op_pg.order, value, rtol=rtol):
         print("Used higher precision, lower precision value", op_pg.order)
         new_optimizer = spatula.optimize.Union.with_step_gradient_descent(
-            spatula.optimize.RandomSearch(max_iter=10000, seed=rng.integers(0, 1000000))
+            spatula.optimize.Mesh.from_lattice(n_rotations=5_000),
+            max_iter=1_000,
+            learning_rate=0.01,
```
Comment (DomFijan) [inline tests/test_pgop.py]: Seems we use learning_rate = 0.01 at several different places. Can we have it as a "global" variable somewhere at the beginning of the file?

**Review by DomFijan — CHANGES_REQUESTED** [review]
Thanks for this. The changes look good. I have 2 comments for consistency, and one possible issue when testing optimizers.

File: tests/test_pgop.py [hunk 2]
```diff
@@ -1510,17 +1510,25 @@ def compute_pgop_polyhedron(
...
+        len(symmetry),
+    )
+    op_pg = compute_op_result(symmetry, OPTIMIZER, mode, system, nlist, sigma, qp)
+    assert op_pg.order.shape == expected_shape
     if not np.allclose(op_pg.order, value, rtol=rtol):
         print("Used higher precision, lower precision value", op_pg.order)
         new_optimizer = spatula.optimize.Union.with_step_gradient_descent(
-            spatula.optimize.RandomSearch(max_iter=10000, seed=rng.integers(0, 1000000))
+            spatula.optimize.Mesh.from_lattice(n_rotations=5_000),
+            max_iter=1_000,
```
[the remark this answers is not in the document]
Comment (janbridley) [reply tests/test_pgop.py]: resolved in #[1fd84de](https://github.com/glotzerlab/spatula/pull/66/commits/1fd84de9a6388985eb02ed1140db311413aa059e) -- looks like 500 works just fine, so I moved it to global
